pkg/bgpv1/manager/reconcilerv2/service_test.go

```diff
@@ -1868,6 +2135,9 @@ func Test_ServiceVIPSharing(t *testing.T) {
 
 		// validate that advertised paths match expected metadata
 		advertisedPrefixesMatch(req, testBGPInstance, tt.expectedMetadata.ServicePaths)
+
+		// validate that advertised policies match expected attributes
+		advertisedPoliciesAttributesMatch(req, testBGPInstance, tt.expectedMetadata.ServiceRoutePolicies)
```

Comment:
@harsimran-pabla I noticed that `advertisedPrefixesMatch()` is only inspecting the metadata cache on `testBGPInstance` but not actually making calls to `testBGPInstance.Router`.  I believe this could lead to inconsistent tests, if there were an issue between the metadata updates and actually configuring GoBGP.

In `advertisedPoliciesAttributesMatch()`, you'll see that I am making calls to `testBGPInstance.Router` for `GetRoutePolicies()` to inspect further.  I wanted to point out the difference here.